src/content/docs/develop/state-management.mdx

```diff
@@ -2,32 +2,144 @@
 
-import Stub from '@components/Stub.astro';
+In a Tauri application, you often need to keep track of the current state of your application or manage the lifecycle of things associated with it. Tauri provides an easy way to manage the state of your application using the `Manager` API, and read it when commands are called.
 
-<Stub>
+Here is a simple example:
 
-Define a state struct and let Tauri manage it:
+```rust
+use tauri::{Builder, Manager};
+
+struct AppData {
+  welcome_message: &'static str,
+}
+
+// In `fn main()`:
+Builder::default()
+  .setup(|app| {
+    app.manage(AppData {
+      welcome_message: "Welcome to Tauri!",
+    });
+    Ok(())
+  })
+  .run(tauri::generate_context!())
+  .unwrap();
+```
+
+You can later access your state via the `App` instance:
 
 ```rust
-use std::{collections::HashMap, sync::Mutex};
-use tauri::{Manager, plugin::Builder};
+let data = app.state::<AppData>();
+```
 
-#[derive(Default)]
-struct Store(Mutex<HashMap<usize, String>>);
+For accessing state in commands, see the [Accessing State](#accessing-state) section.
+
+## Mutability
+
+In Rust, values which can be shared between multiple owners (for example, sent to different threads) cannot be mutable. This is to prevent multiple owners from modifying the value at the same time, which can lead to issues such as data races.
+
+To work around this, you can use the standard library's `Mutex` to wrap your state. This allows you to lock the value when you need to modify it, and unlock it when you are done.
+
+```rust
+use std::sync::Mutex;
 
-Builder::new("<plugin-name>")
-  .setup(|app, api| {
-    app.manage(Store::default());
+use tauri::{Builder, Manager};
 
-    // retrieve the store later
-    let store = app.state::<Store>();
+#[derive(Default)]
+struct AppState {
+  counter: u32,
+}
+
+// In `fn main()`:
+Builder::default()
+  .setup(|app| {
+    app.manage(Mutex::new(AppState::default()));
     Ok(())
   })
+  .run(tauri::generate_context!())
+  .unwrap();
+```
+
+The state can now be modified by locking the mutex:
+
+```rust
+let state = app.state::<Mutex<AppState>>();
+
+// Lock the mutex to get mutable access:
+let mut state = state.lock().unwrap();
+
+// Modify the state:
+state.counter += 1;
+```
+
+At the end of the scope, or when the `MutexGuard` is otherwise dropped, the mutex is unlocked automatically so that other parts of your application can access and mutate the data within.
+
+## Accessing State
+
+You are now storing state in your application, but the obvious next question would be how to access the state in commands.
+
+To build on the previous example, this is how you can access state (and optionally modify it) within commands:
+
+```rust
+use tauri::State;
+
+#[tauri::command]
+fn increase_counter(state: State<'_, Mutex<AppState>>) -> u32 {
+  let mut state = state.lock().unwrap();
+  state.counter += 1;
+  state.counter
+}
+```
+
+For more information on commands, see [Calling Rust from the Frontend](/develop/calling-rust).
+
```

Comment:
Dunno if this small section would do the work @FabianLars, let me know what you think.
```suggestion
### Accessing State Outside Commands

Sometimes you may need to access the state outside of commands, such as in a different thread or in an event handler like on_window_event. In such cases, you can use the state() method of the AppHandle to get the state:

<!-- code snippet -->
rust
use tauri::Manager;

// In an event handler:
fn on_window_event(app: &AppHandle, event: WindowEvent) {
  let state = app.state::<Mutex<AppState>>();
  let mut state = state.lock().unwrap();
  state.counter += 1;
}

This method is useful when you cannot rely on command injection. For example, if you need to move the state into a thread where using an AppHandle is easier, or if you are not in a command context.
```

Reply:
I implemented this, and made some changes after testing it locally